Dockerfile

```diff
@@ -37,9 +37,29 @@ COPY --from=busybox /bin/mkdir /bin/mkdir
 COPY --from=busybox /bin/chmod /bin/chmod
 COPY --from=busybox /bin/cat /bin/cat
 
-# allow anyone to write to this dir, container may not run as root
-RUN mkdir -p /etc/conjur/ssl && chmod 777 /etc/conjur/ssl
+RUN apk add -u shadow libc6-compat && \
+    # Add limited user
+    groupadd -r secrets-provider \
+             -g 777 && \
+    useradd -c "secrets-provider runner account" \
+            -g secrets-provider \
+            -u 777 \
+            -m \
+            -r \
+            secrets-provider && \
+    # Ensure plugin dir is owned by secrets-provider user
+    mkdir -p /usr/local/lib/secrets-provider /etc/conjur/ssl /run/conjur && \
+    # Use GID of 0 since that is what OpenShift will want to be able to read things
+    chown secrets-provider:0 /usr/local/lib/secrets-provider \
+                           /etc/conjur/ssl \
+                           /run/conjur && \
+    # We need open group permissions in these directories since OpenShift won't
+    # match our UID when we try to write files to them
+    chmod 770 /etc/conjur/ssl \
+              /run/conjur
 
-COPY --from=secrets-provider-builder /opt/secrets-provider-for-k8s/secrets-provider /bin
+USER secrets-provider
 
-CMD ["secrets-provider"]
+COPY --from=secrets-provider-builder /opt/secrets-provider-for-k8s/secrets-provider /usr/local/bin/
+
+ENTRYPOINT [ "/usr/local/bin/secrets-provider"]
```

Comment:
why did you change from `CMD` to `ENTRYPOINT`? can you challenge this?

Reply:
CMD sets default command and can be overwritten from command line when docker container runs (with `docker run -it <image> bash`). ENTRYPOINT command and parameters will not be overwritten from command line. So I guess in our case it doesn't really matter but thought I would go with a command that cannot be overwritten

Reply:
I didn't want secrets-provider to be overridden but CMD is safer...